src/Raven.Embedded/RavenEmbedded.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.Text;
+using Raven.Client.Documents;
+using Raven.Client.Exceptions;
+using Raven.Client.Extensions;
+using Raven.Client.ServerWide;
+using Raven.Client.ServerWide.Operations;
+using Sparrow.Logging;
+
+
+namespace Raven.Embedded
+{
+    public class RavenEmbedded
+    {
+        private Process _ravenDbSlavedProcess;
+        private readonly Logger Logger = LoggingSource.Instance.GetLogger<RavenEmbedded>("Embedded");
+        private Task<Uri> _serverTask;
+        private long _initialized;
+
+        private readonly ConcurrentDictionary<string, IDocumentStore> _documentStores = new ConcurrentDictionary<string, IDocumentStore>();
+
+        public void StartServer(ServerOptions options = null)
+        {
+            // Start the server
+
+            if (Interlocked.CompareExchange(ref _initialized, 1, 0) != 0)
+                throw new InvalidOperationException("Already have been initialized");
+
+            options = options ?? ServerOptions.Default;
+            _serverTask = RunServer(options);
+        }
+
+        public static RavenEmbedded Instance = new RavenEmbedded();
+
+        public async Task<IDocumentStore> GetDocumentStore(string database)
```

Comment:
I would suggest adding sync method:
```
public IDocumentStore GetDocumentStore(string database)
```

and rename this one to:
```
public async Task<IDocumentStore> GetDocumentStoreAsync(string database)
```

Reply:
I agree